git-transport/src/client/blocking_io/file.rs

```diff
@@ -282,5 +298,18 @@ mod tests {
                 }
             }
         }
+
+        mod handshake {
+            use crate::client::file::to_single_quoted;
+
+            #[test]
+            fn quoted_strings() {
```

Comment:
I think a couple of tests are missing to [implement this behaviour equivalently](https://github.com/git/git/blob/5dec958dcf965fc75e0f459f8e8ccf9c9f495b15/quote.c#L12-L44). CI seems to fail due to it as well.

Please also move it to [this module](https://github.com/Byron/gitoxide/blob/f7f136dbe4f86e7dee1d54835c420ec07c96cd78/git-quote/src/lib.rs#L6-L5) - in there it's easier to test as well. Try to break it with your tests :).